src/TaskManager/TaskManager/TaskManager.cs

```diff
@@ -250,17 +252,28 @@ private async Task HandleCancellationTask(JsonMessage<TaskCancellationEvent> mes
                     throw new InvalidOperationException("Task Event data not found.");
                 }
 
-                var taskRunner = typeof(ITaskPlugin).CreateInstance<ITaskPlugin>(serviceProvider: _scope.ServiceProvider, typeString: pluginAssembly, _serviceScopeFactory, taskExecEvent);
-                await taskRunner.HandleTimeout(message.Body.Identity);
-
-                AcknowledgeMessage(message);
+                taskRunner = typeof(ITaskPlugin).CreateInstance<ITaskPlugin>(serviceProvider: _scope.ServiceProvider, typeString: pluginAssembly, _serviceScopeFactory, taskExecEvent);
             }
             catch (Exception ex)
             {
                 _logger.UnsupportedRunner(pluginAssembly, ex);
                 await HandleMessageException(message, message.Body.WorkflowInstanceId, message.Body.TaskId, message.Body.ExecutionId, false).ConfigureAwait(false);
                 return;
             }
+
+            try
+            {
+                await taskRunner.HandleTimeout(message.Body.Identity);
+            }
+            catch (Exception ex)
+            {
+                // Ignoring exception here as we've asked for the task to be stopped.
+                _logger.ExectionTimingOutTask(pluginAssembly, ex);
+            }
```

Comment:
I've noticed you're handling the exception for not found within the `HandleTimeout` method, so I'm just wondering if we still want to be doing this here? Or should we be letting the other exceptions bubble up at this point as it would have been previously?

Reply:
changed now, added new exception type for not-founds